scripts/setup_haskell.hs

```diff
@@ -352,3 +332,94 @@ filePathToText = Turtle.format Turtle.fp
 
 textToFilePath :: Text -> FilePath
 textToFilePath = FS.fromText
+
+hvn :: Text
+hvn = "haskell-vim-now"
+
+type HelperRepository = Text
+type HelperTool = Text
+type Resolver = Text
+
+-- install ghcide, write compiler and file type plugin files
+installGhcide :: (MonadIO m, MonadReader HvnConfig m) => m ()
+installGhcide = do
+  hvnCfgDest' <- asks hvnCfgDest
+  gitCloneInstall "https://github.com/digital-asset/ghcide.git" "ghcide" Nothing
+  let vimDir = hvnCfgDest' <> ".vim"
+  writeCompilerFile vimDir
+  writeaFtPluginFile vimDir
+  where
+    mkdir' d = Turtle.testpath d >>= \exists -> unless exists (Turtle.mktree d)
+    toFile f c = liftIO $ Turtle.writeTextFile f c
+    writeaFtPluginFile d = do
+      let d' = d </> "after" </> "ftplugin"
+      mkdir' d'
+      toFile (d' </> "haskell.vim") "compiler ghcide"
+    writeCompilerFile d = do
+      let d' = d <> "compiler"
+      mkdir' d'
+      toFile (d' </> "ghcide.vim") [r|
+CompilerSet errorformat=%-Gghcide\ %s
+CompilerSet errorformat+=%-GReport\ bugs\ at\ %s
+CompilerSet errorformat+=%-GStep\ %s
+CompilerSet errorformat+=%-GFound\ %s
+CompilerSet errorformat+=%-GCradle\ %s
+CompilerSet errorformat+=%-GRange:\ %s
+CompilerSet errorformat+=%-GFile:\ %s
+CompilerSet errorformat+=%-GSource:\ %s
+CompilerSet errorformat+=%-GSeverity:\ %s
+CompilerSet errorformat+=%-GCompleted\ %s
+" exclude empty or whitespace-only lines
+CompilerSet errorformat+=%-G\\s%#
+CompilerSet errorformat+=%EMessage:%\\s%#%>
+CompilerSet errorformat+=%C%\\s%#%f:%l:%c:%\\s%#error:%\\s%#%>
+CompilerSet errorformat+=%C%m
+CompilerSet errorformat+=%ZCompleted%m
+
+setlocal makeprg=ghcide\ %\ 2>&1\ \\\|\ sed\ 's/\\x1B\\[[0-9;]*m//g'
+|]
+
+gitCloneInstall :: MonadIO m => HelperRepository -> HelperTool -> Maybe Resolver -> m ()
+gitCloneInstall repo tool maybeResolver = Turtle.sh $ do
+  liftIO $ msg (unwords' ["clone", repo, "and install", tool])
+  Turtle.mktempdir "/tmp" hvn >>= Turtle.pushd
```

Comment:
The haskell script tried to be compatible with linux and windows, but doing a `/tmp` here won't work on Windows unfortunately.  I'm not convinced worrying about Windows support is really worth it, but some food for thought would be to specify a parent path of somewhere at or under the hvn config directory.

Reply:
`/tmp` issue is fixed in 2c4e163d76c032a7fba81c33445ff0fd249e9761
But since https://github.com/begriffs/haskell-vim-now/pull/297/commits/761e41920cf4e87fa2323e3bf748aad38167403b#diff-8273d963676aedd1589061c6edb52319R379 involved `sed`, Windows wouldn't support  `,hc` key binding.